tests/FSharp.Compiler.ComponentTests/CompilerService/AsyncMemoize.fs

```diff
@@ -101,13 +103,13 @@ let ``Basics``() =
 let ``We can disconnect a request from a running job`` () =
 
     let cts = new CancellationTokenSource()
-    let canFinish = new ManualResetEventSlim(false)
+    let canFinish = new ManualResetEvent(false)
```

Comment:
It used a blocking `Wait` and it was not obvious why it worked at all. I think `Async.AwaitWaitHandle` is better here because it participates in async cancellation.

Reply:
Oh. I thought it was like `SemaphoreSlim` and offered an async option, but looks like it's not.